src/aws.ts

```diff
@@ -0,0 +1,48 @@
+import { WebSocketApi } from "@aws-cdk/aws-apigatewayv2-alpha";
+import {
+  ASLGraph,
+  ErrorCodes,
+  isObjectLiteralExpr,
+  isPropAssignExpr,
+  SynthError,
+} from "functionless";
+import { makeIntegration } from "functionless/lib/integration";
+
+export const InvokeApi = makeIntegration<
+  "$AWS.API.Invoke",
+  (input: {
+    ApiEndpoint: string;
+    Method: string;
+    Stage: string;
+    Path: string;
+    AuthType: string;
+    RequestBody: any;
+  }) => Promise<any>
+>({
+  kind: "$AWS.API.Invoke",
+  asl(call, context) {
+    const input = call.args[0].expr.as(isObjectLiteralExpr);
+    if (!input) throw new Error("Invalid input");
```

Comment:
Looks like you don't need your input to be a literal object. This should work

```ts
return context.evalExprToJsonPathOrLiteral(input, (output) => {
   return context.stateWithHeapOutput(
        ASLGraph.taskWithInput{{
           Type: "Task",
           Resource: "arn:aws:states:::apigateway:invoke",
           Next: ASLGraph.DeferNext,
         }, output);
      });
    });
});
```

Or if you want to get spicier you can validate when it is an object literal.

```ts
return context.evalExprToJsonPathOrLiteral(input, (output) => {
   // if the input is a dynamic reference and not an object, it will fail at runtime
   // typescript will catch this in most cases.
   if(ASLGraph.isLiteralValue(output) && typeof output !== "object") {
       throw new Error("Invalid input");
   }
   return context.stateWithHeapOutput(
        ASLGraph.taskWithInput{{
           Type: "Task",
           Resource: "arn:aws:states:::apigateway:invoke",
           Next: ASLGraph.DeferNext,
         }, output);
      });
    });
});
```


----

If you need your input to be an object literal (which you don't it appears)

nit: you don't need to check if it is an `ObjectLiteralExpr`, just run it through `evalExpr` and then fail if `!ASLGraph.isLiteralValue(output) || typeof output.value !== "object"`.

There is a magical `evalToConstant` function which supports more cases of generating a constant value from the AST and there are a few cases like using dynamic expressions in the objectLiteral which would make it an invalid constant input.

Reply:
Nice, I'll clean this up and make a PR to the functionless repo.